crates/ruff_python_parser/src/semantic_errors.rs

```diff
@@ -118,12 +118,22 @@ impl SemanticSyntaxChecker {
                     );
                 }
             }
-            Stmt::Return(ast::StmtReturn {
-                value: Some(value), ..
-            }) => {
-                // test_err single_star_return
-                // def f(): return *x
-                Self::invalid_star_expression(value, ctx);
+            Stmt::Return(ast::StmtReturn { value, range }) => {
+                if let Some(value) = value {
+                    // test_err single_star_return
+                    // def f(): return *x
+                    Self::invalid_star_expression(value, ctx);
+                }
+                if !self.in_function_scope {
```

Comment:
As stated in the other PR, I'd prefer if we can re-use the scope information from the `Checker`.